aexpect/ops.py

```diff
@@ -424,23 +424,29 @@ def grep(session, expr, path, check=False, flags="-a"):
 ###############################################################################
 
 
-def hash_file(session, filename, method='md5'):
+def hash_file(session, filename, size="", method="md5"):
     """
     Calculate hash of given file from a session.
 
     :param session: session to run the command on
     :type session: ShellSession
     :param str filename: full path of file that should be hashed
+    :param str size: calculate hash from initial data chunk of some size,
+                     e.g. 1M, 1G, etc.
     :param str method: method for hashing
     :returns: hash of file, a hex number
     :rtype: str
     :raises: :py:class:`RuntimeError` if hash command failed (e.g. file not
              found) or resulting output does not have expected format
     """
     cmd = f"{method}sum"
+    if size:
+        cmd = f"head -c {size} {quote(filename)} | md5sum"
```

Comment:
you should use `{cmd}` instead of the `md5sum`

Reply:
Hehe, silly me. Thanks for the nice catch!